base/reflection.jl

```diff
@@ -1083,6 +1083,22 @@ function isambiguous(m1::Method, m2::Method; ambiguous_bottom::Bool=false)
     return true
 end
 
+"""
+    delete_method(m::Method)
+
+Make method `m` uncallable and force recompilation of any methods that use(d) it.
+"""
+function delete_method(m::Method)
+    ccall(:jl_method_table_disable, Void, (Any, Any), MethodTable(m), m)
+end
+
+MethodTable(m::Method) = get_methodtable(m.sig)
+
+get_methodtable(u::UnionAll) = get_methodtable(u.body)
```

Comment:
There's an existing function for this that should be reused.

Reply:
The suspense is killing me, what is it? I changed it to the closest thing I could find, inspired by code in [serialize.jl](https://github.com/JuliaLang/julia/blob/295b0983de731a95ddc63e06bd22fb4a371a9499/base/serialize.jl#L894). There's also [this sequence in methodshow](https://github.com/JuliaLang/julia/blob/295b0983de731a95ddc63e06bd22fb4a371a9499/base/methodshow.jl#L107-L109). I grepped base (`grep -R "\.mt\b" *`) and src (`grep -R DLLEXPORT jl_methtable_t *`) without noticing any better hits.